src/_imaging.c

```diff
@@ -3660,58 +3690,51 @@ static struct PyMethodDef methods[] = {
-    ImagingObject *im_self = ImagingObject_AsStruct(ctx, self);
-    Imaging im = im_self->image;
-    const char  *mode = im->mode;
-    return HPyUnicode_FromString(ctx, mode);
+HPyDef_GET(Imaging_getattr_mode, "mode")
+static HPy Imaging_getattr_mode_get(HPyContext *ctx, HPy self, void *closure) {
+    return HPyUnicode_FromString(ctx, ImagingObject_AsStruct(ctx, self)->image->mode);
 }
 
-HPyDef_GET(Imaging_getattr_size, "size", Imaging_getattr_size_impl)
-static HPy Imaging_getattr_size_impl(HPyContext *ctx, HPy self, void *closure) {
-    ImagingObject *im_self = ImagingObject_AsStruct(ctx, self);
-    return HPy_FromPyObject(ctx, Py_BuildValue("ii", im_self->image->xsize, im_self->image->ysize));
+HPyDef_GET(Imaging_getattr_size, "size")
+static HPy Imaging_getattr_size_get(HPyContext *ctx, HPy self, void *closure) {
+    Imaging image = ImagingObject_AsStruct(ctx, self)->image;
+    return HPy_BuildValue(ctx, "ii", image->xsize, image->ysize);
 }
 
-HPyDef_GET(Imaging_getattr_bands, "bands", Imaging_getattr_bands_impl)
-static HPy Imaging_getattr_bands_impl(HPyContext *ctx, HPy self, void *closure) {
+HPyDef_GET(Imaging_getattr_bands, "bands")
+static HPy Imaging_getattr_bands_get(HPyContext *ctx, HPy self, void *closure) {
     return HPyLong_FromLong(ctx, ((Imaging) ImagingObject_AsStruct(ctx, self))->bands);
 }
 
-HPyDef_GET(Imaging_getattr_id, "id", Imaging_getattr_id_impl)
-static HPy Imaging_getattr_id_impl(HPyContext *ctx, HPy self, void *closure) {
-    return HPyLong_FromSsize_t(ctx, (HPy_ssize_t)((ImagingObject *) HPy_AsPyObject(ctx, self))->image);
+HPyDef_GET(Imaging_getattr_id, "id")
+static HPy Imaging_getattr_id_get(HPyContext *ctx, HPy self, void *closure) {
+    return HPyLong_FromSize_t(ctx, (HPy_ssize_t)ImagingObject_AsStruct(ctx, self)->image);
 }
 
-HPyDef_GET(Imaging_getattr_ptr, "ptr", Imaging_getattr_ptr_impl)
-static HPy Imaging_getattr_ptr_impl(HPyContext *ctx, HPy self, void *closure) {
-    return HPy_FromPyObject(ctx, PyCapsule_New(((ImagingObject *) HPy_AsPyObject(ctx, self))->image, IMAGING_MAGIC, NULL));
+HPyDef_GET(Imaging_getattr_ptr, "ptr")
+static HPy Imaging_getattr_ptr_get(HPyContext *ctx, HPy self, void *closure) {
+    return HPyCapsule_New(ctx, ImagingObject_AsStruct(ctx, self)->image, IMAGING_MAGIC, NULL);
 }
 
-HPyDef_GET(Imaging_getattr_unsafe_ptrs, "unsafe_ptrs", Imaging_getattr_unsafe_ptrs_impl)
-static HPy Imaging_getattr_unsafe_ptrs_impl(HPyContext *ctx, HPy self, void *closure) {
-    Imaging im = ((Imaging) HPy_AsPyObject(ctx, self));
-    return HPy_FromPyObject(ctx, Py_BuildValue(
-        "(sn)(sn)(sn)",
-        "image8",
-        im->image8,
-        "image32",
-        im->image32,
-        "image",
-        im->image8));
+HPyDef_GET(Imaging_getattr_unsafe_ptrs, "unsafe_ptrs")
+static HPy Imaging_getattr_unsafe_ptrs_get(HPyContext *ctx, HPy self, void *closure) {
+    Imaging im = ImagingObject_AsStruct(ctx, self)->image;
```

Comment:
I remember I had an issue here where casting image to Imaging worked, but using ImagingObject_AsStruct didn't compile. If this compiles, it's fine